static/js/lightbox_canvas.js

```diff
@@ -221,6 +221,37 @@ var LightboxCanvas = (function () {
             canvas.width = canvas.width;
             context.imageSmoothingEnabled = false;
 
+            if (canvas.width < canvas.height) {
+                var maxWidth = window.innerWidth * 0.75;
+                var defaultWidth = canvas.width / 2;
+                var adjustedWidth;
+                var ratio;
+
+                // 4 is hardcoded as the zoom level where the maxWidth will be used,
+                // since most people will only be zooming in at the early levels.
+                var maxedOnZoom = 4;
+                var zoomRatio = (meta.zoom - 1) / (maxedOnZoom - 1);
+                if (meta.zoom < maxedOnZoom) {
+                    adjustedWidth = defaultWidth + (maxWidth - defaultWidth) * zoomRatio;
+                    ratio = canvas.width * meta.zoom / adjustedWidth;
+
+                    canvas.style.width = adjustedWidth + "px";
+                    w  = canvas.width / 2 * ratio;
+                    x  = c.x * adjustedWidth * 2 * meta.zoom + adjustedWidth;
+                    if (x >= 0) {x = 0;}
+                    if (x + w <= canvas.width) {x = canvas.width - w;}
+                } else {
+                    adjustedWidth = maxWidth;
+                    ratio = canvas.width * meta.zoom / adjustedWidth;
+
+                    canvas.style.width = adjustedWidth + "px";
+                    w  = canvas.width / 2 * ratio;
+                    x  = c.x * adjustedWidth / 2 * ratio + adjustedWidth;
+                    if (x >= 0) {x = 0;}
+                    if (x + w <= canvas.width) {x = canvas.width - w;}
```

Comment:
This case seems to mostly duplicate the code of the previous case; would it be possible to move most of this code out of the else statement?